dagger/sales.nim

```diff
@@ -0,0 +1,74 @@
+import std/sequtils
+import pkg/questionable
+import pkg/upraises
+import pkg/stint
+import pkg/nimcrypto
+import ./market
+
+export stint
+
+type
+  Sales* = ref object
+    market: Market
+    available*: seq[Availability]
+    subscription: ?Subscription
+
+  Availability* = object
+    id*: array[32, byte]
+    size*: uint64
+    duration*: uint64
+    minPrice*: UInt256
+
+func new*(_: type Sales, market: Market): Sales =
+  Sales(market: market)
+
+proc new*(_: type Availability,
```

Comment:
Nitpick, but per status style guide, we should use `init` when it's an object (and `new` for `ref object`s)